spring/boot-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaClientHttpConnector.java

```diff
@@ -0,0 +1,123 @@
+
+import reactor.core.publisher.Mono;
+
+/**
+ * A {@link ClientHttpConnector} implementation for the Armeria HTTP client.
+ */
+public final class ArmeriaClientHttpConnector implements ClientHttpConnector {
+
+    /**
+     * A default {@link ArmeriaClientHttpConnectorCustomizer} which does nothing.
+     */
+    private static final ArmeriaClientHttpConnectorCustomizer IDENTITY = b -> b;
+
+    private final ArmeriaClientHttpConnectorCustomizer customizer;
+    private final DataBufferFactory bufferFactory;
+
+    /**
+     * Creates an {@link ArmeriaClientHttpConnector} with the default
+     * {@link ArmeriaClientHttpConnectorCustomizer} and {@link DataBufferFactory}.
+     */
+    public ArmeriaClientHttpConnector() {
+        this(IDENTITY, DataBufferFactories.DEFAULT);
+    }
+
+    /**
+     * Creates an {@link ArmeriaClientHttpConnector} with the specified
+     * {@link ArmeriaClientHttpConnectorCustomizer} and the default {@link DataBufferFactory}.
+     *
+     * @param customizer the customizer to be used to build an {@link HttpClient}
+     */
+    public ArmeriaClientHttpConnector(ArmeriaClientHttpConnectorCustomizer customizer) {
+        this(customizer, DataBufferFactories.DEFAULT);
+    }
+
+    /**
+     * Creates an {@link ArmeriaClientHttpConnector}.
+     *
+     * @param customizer the {@link ArmeriaClientHttpConnectorCustomizer} to be used to build an
+     *                   {@link HttpClient}
+     * @param bufferFactory the factory to be used to create a {@link DataBuffer}
+     */
+    public ArmeriaClientHttpConnector(ArmeriaClientHttpConnectorCustomizer customizer,
+                                      DataBufferFactory bufferFactory) {
+        this.customizer = requireNonNull(customizer, "customizer");
+        this.bufferFactory = requireNonNull(bufferFactory, "bufferFactory");
+    }
+
+    @Override
+    public Mono<ClientHttpResponse> connect(
+            HttpMethod method, URI uri, Function<? super ClientHttpRequest, Mono<Void>> requestCallback) {
+        try {
+            final ArmeriaClientHttpRequest request = createRequest(method, uri);
+            return requestCallback.apply(request)
+                                  .then(Mono.fromFuture(request.future()))
+                                  .map(ArmeriaHttpClientResponseSubscriber::new)
+                                  .flatMap(s -> Mono.fromFuture(s.httpHeadersFuture())
+                                                    .map(headers -> createResponse(headers, s)));
+        } catch (IllegalArgumentException e) {
+            return Mono.error(e);
+        }
+    }
+
+    private ArmeriaClientHttpRequest createRequest(HttpMethod method, URI uri) {
+        final String scheme = uri.getScheme();
+        final String authority = uri.getRawAuthority();
+        final String path = uri.getRawPath();
+        final String query = uri.getRawQuery();
+
+        checkArgument(!Strings.isNullOrEmpty(authority), "URI is not absolute: " + uri);
```

Comment:
Do you mean a request path like `http:///api/v4/groups/1`? If so, it will also be failed with the same error message. Actually I wanted to use a message `authority is undefined`, but [this test](https://github.com/spring-projects/spring-framework/blob/master/spring-webflux/src/test/java/org/springframework/web/reactive/function/client/WebClientIntegrationTests.java#L734) checks the message, so I followed it.